File: C++/1800. Maximum Ascending Subarray Sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& n) {
        int s=n[0],m=0;
        for(int i=0;i<n.size()-1;i++){
            if(n[i]<n[i+1]){
                s+=n[i+1];
            }
            else{
                m=max(m,s);
            s=n[i+1];}
        }
        return max(s,m);
    }
};

File: JAVA/1800. Maximum Ascending Subarray Sum.java
class Solution {
    public int maxAscendingSum(int[] n) {
        int s=n[0],m=0;
        for(int i=0;i<n.length-1;i++){
            if(n[i]<n[i+1])
            s+=n[i+1];
            else{
                m=Math.max(m,s);
                s=n[i+1];
            }
        }
        return Math.max(m,s);
    }
}